ts/Data/Connectors/JSONConnector.ts

```diff
@@ -142,7 +142,17 @@ class JSONConnector extends DataConnector {
                     // If already loaded, clear the current rows
                     table.deleteColumns();
                     converter.parse({ data });
-                    table.setColumns(converter.getTable().getColumns());
+
+                    const rowKeysId = table.getRowKeysId();
+                    const columns = converter.getTable().getColumns();
+                    if (rowKeysId) {
+                        columns[rowKeysId] = [];
+                        for (let i = 0; i < data.length; i++) {
+                            columns[rowKeysId][i] = rowKeysId + '_' + String(i);
+                        }
```

Comment:
```suggestion
                        for (let i = 0, iEnd = data.length; i < iEnd; i++) {
                            columns[rowKeysId][i] = rowKeysId + '_' + String(i);
                        }
```

Reply:
Implemented as (otherwise TS complains about 'never' type):
```
            this.columns[id] = [];
            const keysArray = this.columns[id];

```